results/cmd/api/main.go

```diff
@@ -170,16 +180,71 @@ func (s *server) DeleteTaskRunResult(ctx context.Context, req *pb.DeleteTaskRunR
 	return nil, nil
 }
 
+// ListTaskRunsResult receives a ListTaskRunRequest from users and return to users a list of TaskRuns according to the query
+func (s *server) ListTaskRunsResult(ctx context.Context, req *pb.ListTaskRunsRequest) (*pb.ListTaskRunsResponse, error) {
+	// Set up environment for cel and check if filter is empty string
+	ast, issues := s.env.Compile(req.GetFilter())
+	if issues != nil && issues.Err() != nil && req.GetFilter() != "" {
+		log.Printf("type-check error: %s", issues.Err())
+		return nil, status.Errorf(codes.InvalidArgument, "No TaskRuns found for the query string due to wrong type in the query, please try to enter a query with correct type again: %v", issues.Err())
+	}
+	// get all taskruns from database
+	rows, err := s.db.Query("SELECT taskrunlog FROM taskrun")
+	if err != nil {
+		log.Printf("failed to query on database: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse the query string due to internal server error, please try to enter the query again: %v", err)
+	}
+	var taskRunList []*pb.TaskRun
+	for rows.Next() {
+		var taskrunblob []byte
+		if err := rows.Scan(&taskrunblob); err != nil {
+			log.Printf("failed to scan a row in query results: %v", err)
+			return nil, status.Errorf(codes.Internal, "failed to parse the query string due to internal server error, please try to enter the query again: %v", err)
+		}
+		taskrun := &pb.TaskRun{}
+		if err := proto.Unmarshal(taskrunblob, taskrun); err != nil {
+			log.Printf("unmarshaling error: %v", err)
+			return nil, status.Errorf(codes.Internal, "failed to parse the query string due to internal server error, please try to enter the query again: %v", err)
```

Comment:
I'd recommend returning a more detailed error, ideally describing what step is failing. Most users will see the Internal error status and give up, but being able to see a log and see what step failed is very useful when debugging why something went wrong.

Reply:
done